oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtTimestampValidator.java

```diff
@@ -71,6 +77,9 @@ public OAuth2TokenValidatorResult validate(Jwt jwt) {
 
 		if (expiry != null) {
 			if (Instant.now(this.clock).minus(clockSkew).isAfter(expiry)) {
+				if (logger.isDebugEnabled()){
```

Comment:
I see your point that since there is the potential to collect multiple errors, but only one is ultimately in the exception message, that not logging in these places causes an information gap.

What would you think about having the error message and the debug message align? For example, doing something like:

```java
String reason = String.format("Jwt expired at %s", isoDateTime(jwt.getExpiresAt()));
if (logger.isDebugEnabled()) { 
    logger.debug("Failed to validate because " + reason);
}
OAuth2Error error = new OAuth2Error(..., reason, ...);
```

Reply:
Sounds good. I'll reimplement this

Reply:
This looks good for `nbf`. Will you please make the same change for `exp`?